Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp

```diff
@@ -193,6 +197,27 @@ void LoadSpiceXML2DDet::init() {
       new ArrayProperty<size_t>("DetectorGeometry"),
       "A size-2 unsigned integer array [X, Y] for detector geometry. "
       "Such that the detector contains X x Y pixels.");
+
+  declareProperty(
+      "LoadInstrument", true,
+      "Flag to load instrument to output workspace. "
+      "HFIR's HB3A will be loaded if InstrumentFileName is not specified.");
+
+  declareProperty(
+      new FileProperty("InstrumentFilename", "", FileProperty::OptionalLoad,
+                       ".xml"),
+      "The filename (including its full or relative path) of an instrument "
+      "definition file. The file extension must either be .xml or .XML when "
+      "specifying an instrument definition file. Note Filename or "
+      "InstrumentName must be specified but not both.");
```

Comment:
Line 212: should it say `InstrumentFilename` instead of `InstrumentName`? I cannot find `InstrumentName`  in this algorithm.

Also, this mutual exclusion rule could be checked at the beginning of exec(), with an exception thrown to the user if they give both as non-empty values.